tests/std/tests/P0896R4_ranges_alg_partition_copy/test.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <numeric>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+#define ASSERT(...) assert((__VA_ARGS__))
+
+using P = pair<int, int>;
+
+constexpr auto is_even = [](int i) { return i % 2 == 0; };
+
+// Validate that partition_copy_result aliases in_out_out_result
+STATIC_ASSERT(
+    same_as<ranges::partition_copy_result<int, double, void*>, ranges::in_out_out_result<int, double, void*>>);
+
+// Validate dangling story
+template <class T>
+inline constexpr T* nullptr_to = nullptr;

```

Comment:
That one is neat. Should we move it to range algorithm suppor so that we can reuse it for the other tests as well?

Reply:
Sure - why not?